docs/source/jit_language_reference_v2.rst

```diff
@@ -0,0 +1,320 @@
+
+::
+
+    shift_expr ::=  a_expr | shift_expr ( '<<' | '>>' ) a_expr
+
+
+These operators accept two ``int`` arguments, two ``Tensor`` arguments, or a ``Tensor`` argument and an ``int`` or
+``float`` argument. In all cases, a right shift by ``n`` is defined as floor division by ``pow(2, n)`` and a left shift
+by ``n`` is defined as multiplication by ``pow(2, n)``. When both arguments are ``Tensors``, they must have the same
+shape. When one is a scalar and the other is a ``Tensor``, the scalar is logically broadcasted to match the size of
+the ``Tensor``.
+
+Binary Bitwise Operations
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+::
+
+    and_expr ::=  shift_expr | and_expr '&' shift_expr
+    xor_expr ::=  and_expr | xor_expr '^' and_expr
+    or_expr  ::=  xor_expr | or_expr '|' xor_expr
+
+
+The ``&`` operator computes the bitwise AND of its arguments, the ``^`` the bitwise XOR and ``|`` the bitwise OR.
+Both operands must be ``int`` or ``Tensor``, or the left operand must be ``Tensor`` and the right operand must be
+``int``. When both operands are ``Tensor``, they must have the same shape. When the right operand is ``int``, and
+the left operand is ``Tensor`` , the right operand is logically broadcasted to match the shape of the ``Tensor``.
+
+Comparisons
+^^^^^^^^^^^
+
+::
+
+    comparison    ::=  or_expr (comp_operator or_expr)*
+    comp_operator ::=  '<' | '>' | '==' | '>=' | '<=' | '!=' | 'is' ['not'] | ['not'] 'in'
+
+A comparison yields a boolean values (``True`` or ``False``) or, if one of the operands is a ``Tensor``, a boolean
+``Tensor``. Comparisons can be chained arbitrarily as long as they do not yield boolean ``Tensors`` that have more
+than one element. ``a op1 b op2 c ...`` is equivalent to ``a op1 b and b op2 c and ...``.
+
+Value Comparisons
+"""""""""""""""""
+The operators ``<``, ``>``, ``==``, ``>=``, ``<=``, and ``!=`` compare the values of two objects. The two objects generally need to be of
+the same type, unless there is an implicit type conversion available between the objects. User-defined types can
+be compared if rich comparison methods ( ``__lt__`` etc.) are defined on them. Built-in type comparison works like
+Python:
+
+* numbers are compared mathematically
+* strings compare lexicographically
+* lists, tuples, and dicts can be compared only to other lists, tuples, and dicts of the same type and are compared using the comparison operator of corresponding elements
+
+Membership Test Operations
+""""""""""""""""""""""""""
+The operators ``in`` and ``not in`` test for membership. ``x in s`` evaluates to ``True`` if ``x`` is a member of ``s`` and ``False```` otherwise.
+``x not in s`` is equivalent to ``not x in s``. This operator is supported for lists, dicts, and tuples, and can be used with
+user-defined types if they implement the ``__contains__`` method.
+
+Identity Comparisons
+""""""""""""""""""""
+For any types stored inside ``IValue`` as pointers, operators ``is`` and ``is not`` test for the object’s identity;
```

Comment:
It might be unclear to users as to which types are stored as pointers in IValues, maybe we can attach a list here for refernece?